projects/step-core/src/lib/modules/plan-common/components/plan-create-dialog/plan-create-dialog.component.ts

```diff
@@ -29,6 +33,35 @@ export class PlanCreateDialogComponent {
 
   readonly artefactTypes$ = this._api.getArtefactTemplates().pipe(shareReplay(1));
 
+  filterMultiControl: FormControl<string | null> = new FormControl<string>('');
+  filterMultiTypeControl: FormControl<string | null> = new FormControl<string>('');
+  dropdownItemsFiltered: ItemInfo[] = [];
+  dropdownItemsTypeFiltered$: Observable<string[]> | undefined;
+
+  ngAfterContentInit(): void {
+    this.dropdownItemsFiltered = [...this._planTypes];
+    this.filterMultiControl.valueChanges.pipe(takeUntilDestroyed(this._destroyRef)).subscribe((value) => {
+      if (value) {
+        this.dropdownItemsFiltered = this._planTypes.filter((item) =>
+          item.label.toLowerCase().includes(value.toLowerCase()),
+        );
+      } else {
+        this.dropdownItemsFiltered = [...this._planTypes];
+      }
+    });
+
+    this.dropdownItemsTypeFiltered$ = this.artefactTypes$.pipe(share());
+    this.filterMultiTypeControl.valueChanges.pipe(takeUntilDestroyed(this._destroyRef)).subscribe((value) => {
+      if (value) {
+        this.dropdownItemsTypeFiltered$ = this.artefactTypes$.pipe(
+          map((list) => list.filter((item) => item.toLowerCase().includes(value.toLowerCase()))),
+        );
+      } else {
+        this.dropdownItemsTypeFiltered$ = this.artefactTypes$.pipe(share());
+      }
+    });
+  }
```

Comment:
Observable created in the subscription of another observable.
It's not clear when `dropdownItemsTypeFilter$` will be destroyed.
Please organize it in a single observable chain (Check the commend above).
Also in current case the `combineLatest` will be required.